src/background/services/keyAutoAdd.ts

```diff
@@ -137,15 +143,20 @@ export class KeyAutoAddService {
           ),
         );
       } else if (message.action === 'PROGRESS') {
-        // can also save progress to show in popup
-        const currentStep = this.getCurrentStep(message.payload.steps);
+        const steps = message.payload.steps;
+        const timeoutIn = steps
+          .filter(({ status }) => status === 'pending' || status === 'active')
+          .reduce((acc, { maxDuration }) => acc + maxDuration, BASE_TIMEOUT);
+        timeout.reset(timeoutIn);
+        const currentStep = this.getCurrentStep(steps);
         if (currentStep) {
           this.setConnectState(currentStep.name);
         }
         for (const p of ports) {
           if (p !== port) p.postMessage(message);
         }
       } else {
+        removeListeners();
```

Comment:
This was missing again. Also, no harm if we remove already removed listeners. Better safe than sorry.